Sources/Starknet/Accounts/StarknetAccount.swift

```diff
@@ -151,35 +151,35 @@ public class StarknetAccount: StarknetAccountProtocol {
         let params = StarknetInvokeParamsV3(nonce: nonce, l1ResourceBounds: resourceBounds.l1Gas)
         let signedTransaction = try signV3(calls: calls, params: params, forFeeEstimation: false)
 
-        return try await provider.addInvokeTransaction(signedTransaction)
+        return try await provider.addInvokeTransaction(signedTransaction).send()
     }
 
     public func estimateFeeV1(calls: [StarknetCall], nonce: Felt, skipValidate: Bool) async throws -> StarknetFeeEstimate {
         let params = StarknetInvokeParamsV1(nonce: nonce, maxFee: .zero)
         let signedTransaction = try signV1(calls: calls, params: params, forFeeEstimation: true)
 
-        return try await provider.estimateFee(for: signedTransaction, simulationFlags: skipValidate ? [.skipValidate] : [])
+        return try await provider.estimateFee(for: signedTransaction, simulationFlags: skipValidate ? [.skipValidate] : []).send()[0]
```

Comment:
should be return [0] here?

Reply:
Yes, please look at the [old implementation](https://github.com/software-mansion/starknet.swift/pull/187/files#diff-38b6912cf986d0587267d2fcef717a429f3df0977ca75721555cada603ca4bffL216) of calling `estimateFee`. We could directly access value at index 0, because the returned type [here](https://github.com/software-mansion/starknet.swift/pull/187/files#diff-38b6912cf986d0587267d2fcef717a429f3df0977ca75721555cada603ca4bffL184) was `[StarknetFeeEstimate]`. Now it's `Request<[StarknetFeeEstimate], EstimateFeeParams>`, therefore first we need to send and then access [0] element.